src/read_pool.rs

```diff
@@ -44,14 +90,30 @@ impl ReadPool {
 
                 pool.spawn(move || Box::pin(f.never_error()).compat())?;
             }
-            ReadPool::Yatp(remote) => {
+            ReadPoolHandle::Yatp {
+                remote,
+                running_tasks,
+                max_tasks,
+            } => {
+                let running_tasks = running_tasks.clone();
+                if running_tasks.get() as usize >= *max_tasks {
+                    return Err(ReadPoolError::UnifiedReadPoolFull);
+                }
+
                 let fixed_level = match priority {
                     CommandPri::High => Some(0),
                     CommandPri::Normal => None,
                     CommandPri::Low => Some(2),
                 };
                 let extras = Extras::new_multilevel(task_id, fixed_level);
-                let task_cell = TaskCell::new(f, extras);
+                let task_cell = TaskCell::new(
+                    async move {
+                        running_tasks.inc();
```

Comment:
It's possible the running tasks count exceeds max tasks.

Reply:
Do you mean that several requests are spawned at the same time and pass the check in L99? The limit is not strict. Exceeding it a small bit does not matter because the limit tends to be far greater than the concurrency number.
I add a comment for it.